src/frontend/screens/Observation/Buttons.tsx

```diff
@@ -102,29 +102,52 @@ export const ButtonFields = ({
   }
 
   async function handlePressShare() {
-    const {lon, lat} = observation;
+    const {lon, lat, attachments} = observation;
     setShareButtonLoading(true);
+    const getValidUrls = (queries: typeof attachmentUrlQueries) => {
+      const urls = queries
+        .map(query => query.data?.url)
+        .filter((url): url is string => url !== undefined && url !== null);
 
-    const urlsQueries = await Promise.all(
-      attachmentUrlQueries.map(q => q.refetch()),
-    );
-    const urls = urlsQueries.map(query => query.data!.url);
-    const base64Urls = await Promise.all(
-      urls.map(url => convertUrlToBase64(url)),
-    );
+      return urls;
+    };
 
-    Share.open({
-      title: base64Urls.length > 0 ? t(m.shareMediaTitle) : t(m.shareTextTitle),
-      urls: base64Urls,
-      message: t(m.shareMessage, {
-        category_name: preset.name,
-        date: Date.now(),
-        time: Date.now(),
-        coordinates: lon && lat ? formatCoords({lon, lat, format}) : '',
-      }),
-    })
-      .catch(() => {})
-      .finally(() => setShareButtonLoading(false));
+    let urls: string[] = [];
+
+    if (attachments.length > 0) {
+      urls = getValidUrls(attachmentUrlQueries);
+
+      if (urls.length === 0) {
+        setShareButtonLoading(false);
+        Alert.alert('Error', 'Unable to share this observation.');
+        return;
+      }
+    }
+
+    try {
+      const base64Urls = await Promise.all(
+        urls.map(url => convertUrlToBase64(url as string)),
```

Comment:
No need for the type coercion

```suggestion
        urls.map(url => convertUrlToBase64(url)),
```